### PR TITLE
[Merged by Bors] - feat: use junk value in the definition of conditionally complete linear order

### PR DESCRIPTION
**Pull request by sgouezel**
Currently, in a conditionally complete linear order, the supremum of an unbounded set hasn't any specific property. However, in all instances in mathlib, all unbounded sets have the same supremum. This PR adds this requirement in mathlib. This will be convenient to remove boundedness assumptions in measurability statements.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 1]
```diff
@@ -961,6 +979,52 @@ theorem exists_lt_of_ciInf_lt [Nonempty ι] {f : ι → α} (h : iInf f < a) : 
   @exists_lt_of_lt_ciSup αᵒᵈ _ _ _ _ _ h
 #align exists_lt_of_cinfi_lt exists_lt_of_ciInf_lt
 
+theorem csSup_of_not_bddAbove {s : Set α} (hs : ¬BddAbove s) : sSup s = sSup univ :=
+  ConditionallyCompleteLinearOrder.csSup_of_not_bddAbove s hs
+
+theorem csInf_of_not_bddBelow {s : Set α} (hs : ¬BddBelow s) : sInf s = sInf univ :=
+  ConditionallyCompleteLinearOrder.csInf_of_not_bddBelow s hs
```
Comment (eric-wieser) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:986]: Are there any other places in mathlib where these two lemmas can be used to simplify a proof?

**Review by eric-wieser — APPROVED** [review]
bors d+

It's not clear to me whether this really gains much (maybe that's coming in a future PR), but the new fields in the typeclass are at worst harmless.

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean (hunk 1 above)
Comment (sgouezel) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:985]: Yes, even to simplify statements. Coming later.

**Comment by sgouezel** [comment]
bors r+
Thanks for the review!

**Pull request closed without merging**